disp_driver/src/display.cpp

```diff
@@ -0,0 +1,90 @@
+#include "display.hpp"
+
+#include "esp_log.h"
+#include "esp_err.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "esp_timer.h"
+
+#include "lvgl.h"
+#include "lvgl_helpers.h"
+
+void Display::lvgl_timer_task(void *arg)
+{
+    lv_tick_inc(CONFIG_LVGL_TICK_RATE);
+}
+
+void Display::lvgl_gui_task(void *arg)
+{
+    ESP_LOGD(TAG, "LVGL GUI task started");
+
+#ifdef CONFIG_DISPLAY_FRAME_BUFFER_PSRAM
+    static lv_color_t *buf1 = (lv_color_t *)heap_caps_malloc(DISP_BUF_SIZE * sizeof(lv_color_t), MALLOC_CAP_SPIRAM);
+    static lv_color_t *buf2 = (lv_color_t *)heap_caps_malloc(DISP_BUF_SIZE * sizeof(lv_color_t), MALLOC_CAP_SPIRAM);
+#else
+    static lv_color_t *buf1 = (lv_color_t *)heap_caps_malloc(DISP_BUF_SIZE * sizeof(lv_color_t), MALLOC_CAP_INTERNAL);
+    static lv_color_t *buf2 = (lv_color_t *)heap_caps_malloc(DISP_BUF_SIZE * sizeof(lv_color_t), MALLOC_CAP_INTERNAL);
+#endif

```

Comment:
Very nice

Reply:
Kind of a shame that IDF doesn't allow for safer programming in this version. Maybe we can have a separate backlog for "far in the future" items like allocating these at compile time?